### PR TITLE
Allow for different head url

### PR DESCRIPTION
**Pull request by markkurt**

### CODE REVIEW

File: lib/core/TruncateAsyncTask.js [hunk 1]
```diff
@@ -9,11 +9,17 @@ var _execute = function(callback) {
 	
 	this.callback = callback;
 	var self = this;
-	fs.ftruncate(this.fd, this.fileSize, self.callback);
+  //support older versions of node
+  if (fs.ftruncate) {
+    fs.ftruncate(this.fd, this.fileSize, self.callback);
+  }
+  else {
+    fs.truncate(this.fd, this.fileSize, self.callback);
```
Comment (tusharmath) [inline lib/core/TruncateAsyncTask.js]: Truncate takes the file path isn't it as the first param?

**Comment by tusharmath** [comment]
Why would someone have a different HEAD url ?

File: lib/core/TruncateAsyncTask.js (hunk 1 above)
Comment (markkurt) [inline lib/core/TruncateAsyncTask.js]: I believe older versions of node don't have ftruncate, but do allow passing the file descriptor.  I actually didn't mean to create a PR against your repository, I meant to create it against the fork.   Happy to resubmit if you want the changes.

**Comment by markkurt** [comment]
I am using signed S3 URLs.  I haven't found a library that allows you to sign a URL for both HEAD and GET, which means you end up with two different URLs.  One solution was http://stackoverflow.com/questions/15717230/pre-signing-amazon-s3-urls-for-both-head-and-get-verbs, however my use case is a little more complicated.  Having an addition HEAD url passed in the params seemed like the cleanest solution.

**Pull request closed without merging**